bescort.lic

```diff
@@ -599,26 +612,26 @@ class Bescort
       exit
     end
 
-    source_shard = DRRoom.room_objs.find{|obj| obj =~ Regexp.union(mode_to_shard.values)}
-                                   .match(Regexp.union(mode_to_shard.values))
-    if source_shard.nil?
+    unless room_id_to_shard.keys.include?(Room.current.id)
       echo "You are not in a room with a Grazhir shard."
       exit
     end
 
-    fput("prep mg")
-    fput("focus #{source_shard}")
-    waitfor 'You feel fully prepared to cast your spell.'
-    3.times{ fput("harness 20") }
-    fput("cast #{source_shard}")
+    source_shard = room_id_to_shard[Room.current.id]
+    prepare?('mg', 5, symbiosis = false, command='prepare')
+    bput("focus #{source_shard}", 'Roundtime')
+    waitcastrt?
+    3.times{ bput("harness 20", 'You tap') }
+    bput("cast #{source_shard}", 'You are pulled through your unstable Moongate')
```

Comment:
I encourage you to use the `cast` helper in common-arcana (and add the 'pulled through' message to base-spells).